tests/onnx/models.py

```diff
@@ -306,5 +338,304 @@ def __init__(self):
+            inputs=["relu_X", "softmax_1"],
+            outputs=["Y"],
+        )
+
+        graph_def = onnx.helper.make_graph(
+            nodes=[relu_x_node, softmax_node, mul_node],
+            name="Net",
+            inputs=[X],
+            outputs=[Y],
+        )
+
+        op = onnx.OperatorSetIdProto()
+        op.version = OPSET_VERSION
+        model = onnx.helper.make_model(graph_def, opset_imports=[op])
+        onnx.checker.check_model(model)
+        super().__init__(model, [input_shape], 'one_input_port_quantizable_model.dot')
+
+
+class ManyInputPortsQuantizableModel(ONNXReferenceModel):
+    #             X
+    #             |
+    #            ReLU
+    #           /  |
+    #    Identity  |
+    #       |    \  |
+    #    Softmax  Mul
+    #             |
+    #             Y
+    def __init__(self):
+        input_shape = output_shape = [1, 1, 5, 5]
+
+        # IO tensors (ValueInfoProto).
+        model_input_name = "X"
+        X = onnx.helper.make_tensor_value_info(model_input_name,
+                                               onnx.TensorProto.FLOAT,
+                                               input_shape)
+        model_output_name = "Y"
+        Y = onnx.helper.make_tensor_value_info(model_output_name,
+                                               onnx.TensorProto.FLOAT,
+                                               output_shape)
+
+        model_output_name1 = "Y1"
+        Y1 = onnx.helper.make_tensor_value_info(model_output_name1,
+                                                onnx.TensorProto.FLOAT,
+                                                output_shape)
+
+        relu_x_node = onnx.helper.make_node(
+            name="Relu",
+            op_type='Relu',
+            inputs=["X"],
+            outputs=["relu_X"],
+        )
+
+        identity_node = onnx.helper.make_node(
+            name="Identity",
+            op_type="Identity",
+            inputs=["X"],
+            outputs=["identity_1"]
+        )
+
+        softmax_node = onnx.helper.make_node(
+            name="Softmax",
+            op_type="Softmax",
+            inputs=["identity_1"],
+            outputs=["Y1"]
+        )
+
+        mul_node = onnx.helper.make_node(
+            name='Mul',
+            op_type="Mul",
+            inputs=["relu_X", "identity_1"],
+            outputs=["Y"],
+        )
+
+        graph_def = onnx.helper.make_graph(
+            nodes=[relu_x_node, identity_node, softmax_node, mul_node],
+            name="Net",
+            inputs=[X],
+            outputs=[Y, Y1],
+        )
+
+        op = onnx.OperatorSetIdProto()
+        op.version = OPSET_VERSION
+        model = onnx.helper.make_model(graph_def, opset_imports=[op])
+        onnx.checker.check_model(model)
+        super().__init__(model, [input_shape], 'many_input_ports_quantizable_model.dot')
+
+
+ALL_MODELS = [ManyInputPortsQuantizableModel, OneInputPortQuantizableModel, WeightSharingModel, ReshapeWeightModel,
```

Comment:
Don't do registration in this way, otherwise one might forget to add a model to this list after defining the class. You may instead use a decorator to register each class at the same point where it is defined.

Reply:
Done

Reply:
Consider reusing `nncf.common.utils.registry.Registry` which supplies a decorator already written for you.

Reply:
In what sense it is an overhead for *you*? You should take into account overheads to overall code base, or to execution time, not a *personal* overhead. 😕 

You can use `*registry_name*.registry_dict.values()` to iterate over values. Or, better yet, since you noticed a good use case for iteration over registry values throughout the entire NNCF, just add the corresponding helper method or an iterator protocol to the `Registry` class.

Reply:
After discussion - we made a conclusion that it makes sense to utilize `Registry`, while it needs to update this class.